sdcm/utils/loader_utils.py

```diff
@@ -87,29 +74,30 @@ def _run_all_stress_cmds(self, stress_queue, params):
             stress_cmds *= stress_multiplier
 
         for stress_cmd in stress_cmds:
-            params.update({'stress_cmd': stress_cmd})
-            self._parse_stress_cmd(stress_cmd, params)
+            stress_params = dict(params)
+            stress_params.update({'stress_cmd': stress_cmd})
+
+            # Due to an issue with scylla & cassandra-stress - we need to create the counter table manually
+            if 'counter_' in stress_cmd:
+                self._create_counter_table()
+
+            if 'compression' in stress_cmd:
+                if 'keyspace_name' not in stress_params:
+                    compression_prefix = re.search('compression=(.*)Compressor', stress_cmd).group(1)
+                    keyspace_name = "keyspace_{}".format(compression_prefix.lower())
+                    stress_params.update({'keyspace_name': keyspace_name})
 
             # Run all stress commands
             self.log.debug('stress cmd: {}'.format(stress_cmd))
             if stress_cmd.startswith('scylla-bench'):
                 stress_queue.append(self.run_stress_thread(stress_cmd=stress_cmd,
                                                            stats_aggregate_cmds=False,
                                                            round_robin=self.params.get('round_robin')))
-            elif stress_cmd.startswith('cassandra-harry'):
-                stress_queue.append(self.run_stress_thread(**params))
             else:
-                stress_queue.append(self.run_stress_thread(**params))
+                stress_queue.append(self.run_stress_thread(**stress_params))
 
             time.sleep(10)
 
-            # Remove "user profile" param for the next command
-            if 'profile' in params:
-                del params['profile']
-
-            if 'keyspace_name' in params:
```

Comment:
How this is a fix ?

The multiple keyspaces would be broken because of this

Reply:
this is actual fix - removing `keyspace_name` from params was causing the issue. 
Now in case of stress params modifications we anyway use copy of original params for next iteration